app/src/main/java/com/sopt/smeem/data/usecase/MySummaryUseCase.kt

```diff
@@ -0,0 +1,27 @@
+package com.sopt.smeem.data.usecase
+
+import com.sopt.smeem.domain.dto.MyPlanDto
+import com.sopt.smeem.domain.dto.MySmeemDataDto
+import com.sopt.smeem.domain.repository.UserRepository
+import kotlinx.coroutines.coroutineScope
+import javax.inject.Inject
+
+class MySummaryUseCase @Inject constructor(
+    private val userRepository: UserRepository
+) {
+
+    // TODO : 배지까지 반영 -> 주석 해제
+//    suspend operator fun invoke(): Triple<MySmeemDataDto, MyPlanDto?, List<GetBadgeListDto>> =
+//        coroutineScope {
+//            val smeemData = userRepository.getMySmeemData().data()
+//            val myPlan = userRepository.getMyPlanData().data()
+//            val badgeList = userRepository.getMyBadges().data()
+//            Triple(smeemData, myPlan, badgeList)
+//        }
+
+    suspend operator fun invoke(): Pair<MySmeemDataDto, MyPlanDto?> = coroutineScope {
+        val smeemData = userRepository.getMySmeemData().data()
+        val myPlan = userRepository.getMyPlanData().data()
+        Pair(smeemData, myPlan)
```

Comment:
```suggestion
        val smeemData = async { userRepository.getMySmeemData() }
        val myPlan = async { userRepository.getMyPlanData() }
        Pair(smeemData.await().data(), myPlan.await().data())
```

이렇게 해야 non-blocking 하게 가져올듯요

Reply:
> 이렇게 해야 non-blocking 하게 가져올듯요

이렇게 하면 한 비동기 작업이 끝나면 Pair 값 바로 방출해서 뷰가 바로 보일 것 같은데 어떻게 생각하시나요?
배지 통신 부분까지 생기면 세 작업이 모두 끝나야만 뷰를 보여줄건데 async/await를 사용하면 개별적으로 처리될 것 같슴다